src/main/java/eladkay/scanner/terrain/TileEntityTerrainScanner.java

```diff
@@ -75,72 +78,74 @@ public void update() {
+                }
+            }
+            Oregistry.getEntryList().stream().filter(entry -> current.getY() < entry.maxY && current.getY()> entry.minY).forEach(entry -> {
+                int i = ThreadLocalRandom.current().nextInt(entry.rarity);
+                if (i == 0) worldObj.setBlockState(current, entry.ore, 2);
+            });
         }
-        if (y >= 256) {
-            x = pos.getX();
-            y = -1;
-            z = pos.getZ();
+
+        //Movement needs to happen BELOW oregen else things get weird and desynced
+        current.move(EnumFacing.EAST); //X++
+        BlockPos end = this.getEnd(); //We do this lazy load do it can cache the right value
+
+        if (current.getX() > end.getX())
+            current.setPos(pos.getX() + 1, current.getY(), current.getZ() + 1);
+        if (current.getZ() > end.getZ())
+            current.setPos(current.getX(), current.getY() + 1, pos.getZ());
+        if (current.getY() > end.getZ())
+        {
+            current.setPos(pos.getX() + 1, -1, pos.getZ());
             on = false;
-        } else if(y >= 64) on = false;
-        if(worldObj.isRemote) return;
-        if(Config.genVanillaOres) {
-            if (worldObj.getBlockState(new BlockPos(x, y, z)).getBlock() != Blocks.STONE)
-                return;
-            if (y > 8) {
-                int i = ThreadLocalRandom.current().nextInt(25);
-                if (i == 0)
-                    worldObj.setBlockState(new BlockPos(x, y, z), Blocks.COAL_ORE.getDefaultState(), 2);
-                else if (i == 1)
-                    worldObj.setBlockState(new BlockPos(x, y, z), Blocks.IRON_ORE.getDefaultState(), 2);
-            }
-            if (y > 8 && y < 16) {
-                int i = ThreadLocalRandom.current().nextInt(150);
-                if (i == 0)
-                    worldObj.setBlockState(new BlockPos(x, y, z), Blocks.DIAMOND_ORE.getDefaultState(), 2);
-                else if (i == 1)
-                    worldObj.setBlockState(new BlockPos(x, y, z), Blocks.EMERALD_ORE.getDefaultState(), 2);
-                else if (i == 2)
-                    worldObj.setBlockState(new BlockPos(x, y, z), Blocks.REDSTONE_ORE.getDefaultState(), 2);
-                else if (i == 3)
-                    worldObj.setBlockState(new BlockPos(x, y, z), Blocks.LAPIS_ORE.getDefaultState(), 2);
-            }
-            if (y > 8 && y < 32) {
-                int i = ThreadLocalRandom.current().nextInt(45);
-                if (i == 0)
-                    worldObj.setBlockState(new BlockPos(x, y, z), Blocks.GOLD_ORE.getDefaultState(), 2);
-            }
         }
-        Oregistry.getEntryList().stream().filter(entry -> y < entry.maxY && y > entry.minY).forEach(entry -> {
-            int i = ThreadLocalRandom.current().nextInt(entry.rarity);
-            if (i == 0) worldObj.setBlockState(new BlockPos(x, y, z), entry.ore, 2);
-        });
+        else if (current.getY() > 64)
```

Comment:
It's used for a BlockState, it was kind of broken, sort of, but it did work and it was used